gapic/src/main/com/google/gapid/perfetto/models/Selection.java

```diff
@@ -17,95 +17,140 @@
 
 import static com.google.gapid.util.MoreFutures.transform;
 import static com.google.gapid.util.MoreFutures.transformAsync;
-import static java.util.Arrays.stream;
-import static java.util.stream.Collectors.joining;
 
-import com.google.common.collect.Iterables;
 import com.google.common.collect.Maps;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gapid.perfetto.models.SliceTrack.Slice;
+import com.google.gapid.perfetto.models.ThreadTrack.StateSlice;
 import com.google.gapid.perfetto.views.MultiSelectionView;
 import com.google.gapid.perfetto.views.State;
 
+import org.eclipse.swt.SWT;
 import org.eclipse.swt.widgets.Composite;
 
+import java.util.Iterator;
 import java.util.Map;
+import java.util.NavigableMap;
+import java.util.TreeMap;
 
 /**
  * Data about the current selection in the UI.
  */
-public interface Selection {
+public interface Selection<Key> {
   public String getTitle();
+  public boolean contains(Key key);
   public Composite buildUi(Composite parent, State state);
-  public default void mark(@SuppressWarnings("unused") State state) { /* do nothing */ }
+  public default void markTime(@SuppressWarnings("unused") State state) { /* do nothing */ }
   public default void zoom(@SuppressWarnings("unused") State state) { /* do nothing */ }
 
-  public static class MultiSelection implements Selection {
-    private final Selection[] selections;
+  public static class MultiSelection {
+    private final NavigableMap<Kind<?>, Selection<?>> selections;
 
-    public MultiSelection(Selection[] selections) {
-      this.selections = selections;
+    public <Key> MultiSelection(Kind<Key> type, Selection<Key> selection) {
+      this.selections = Maps.newTreeMap();
+      this.selections.put(type, selection);
     }
 
-    @Override
-    public String getTitle() {
-      return stream(selections).map(Selection::getTitle).collect(joining(", "));
+    public MultiSelection(NavigableMap<Kind<?>, Selection<?>> selections) {
+      this.selections = selections;
     }
 
-    @Override
     public Composite buildUi(Composite parent, State state) {
-      if (selections.length == 1) {
-        return selections[0].buildUi(parent, state);
+      if (selections.size() == 1) {
+        return firstSelection().buildUi(parent, state);
       } else {
         return new MultiSelectionView(parent, selections, state);
       }
     }
 
-    @Override
-    public void mark(State state) {
+    public <Key> Selection<Key> getSelection(Kind<Key> type) {
+      return selections.containsKey(type) ? (Selection<Key>) selections.get(type) :type.EmptySelection();
```

Comment:
Add the `@SupressWarnign` annotation here, since this should warn about the cast.

(Also small ws nit after the `:`)